cpp/src/io/comp/statistics.cu

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "gpuinflate.hpp"
+
+#include <rmm/exec_policy.hpp>
+#include <thrust/transform_reduce.h>
+
+namespace cudf::io {
+
+writer_compression_statistics collect_compression_statistics(
+  device_span<device_span<uint8_t const> const> inputs,
+  device_span<compression_result const> results,
+  rmm::cuda_stream_view stream)
+{
+  // bytes_written on success
+  auto const output_size_successful = thrust::transform_reduce(
+    rmm::exec_policy(stream),
+    results.begin(),
+    results.end(),
+    [] __device__(auto& res) {
```

Comment:
We could include `const` here, but it wouldn't have any impact because results is a `device_span<compression_result const>` and its elements are always accessed as `const`. So in the current code `auto` is deduced as `compression_result const`. With your change it would be deduced as `compression_result`, but the type of `res` does not change.
Seems like personal preference, let me know what you think.